plugins/headlamp-plugin/package.json

```diff
@@ -108,6 +111,7 @@
     "monaco-editor-webpack-plugin": "^4.2.0",
     "msw": "^0.47.4",
     "msw-storybook-addon": "^1.10.0",
+    "nock": "^13.5.4",
```

Comment:
Can you please explain why nock is used over other options?

Reply:
it's much simpler to set up than, say, MSW, with comparable if not better performance at scale. but I imagine either would work

Reply:
Ok, cool. Sounds like we should use nock then.

---

I looked into it a bit further and here are some notes in case anyone else is interested about the packages added.

I see we're already sort of msw in the storybook files. But only to mock some of the calls that some components are making. So I'd like to know if there is a good reason to add a new dependency. But also evaluate each dependency.

https://mswjs.io/docs/comparison
https://github.com/nock/nock
https://github.com/mswjs/msw
https://securityscorecards.dev/viewer/?uri=github.com/nock/nock
https://securityscorecards.dev/viewer/?uri=github.com/mswjs/msw
https://www.npmjs.com/package/nock
https://www.npmjs.com/package/msw

The tests are already using nock, so avoiding rewrite work would be a plus. I'm not sure how complicated a rewrite it would be(chuck it into copilot and it's done?).

It appears we couldn't use nock for the in-browser tests, because it only works on node. But saying that it appears our msw setup in storybook is at least partially broken. 

Also we're using a pretty old msw 0.47.4 and it's up to 2.x already. So we'd need to upgrade that. Except https://github.com/mswjs/msw-storybook-addon only supports < 2.0 at the moment. This means we can't use msw here at the moment without taking on the need to rewrite it later.

https://www.npmjs.com/package/ws
https://github.com/websockets/ws
https://securityscorecards.dev/viewer/?uri=github.com/websockets/ws

Reply:
it would make sense to me if we wanted to stay consistent and use msw throughout (it's more versatile than nock, anyway) and I don't think a refactor in msw would be too difficult like you mention, but performance-wise nock may be better down the road

here's a comparison between the two:
https://www.bam.tech/article/nock-vs-msw-i-tested-both-and-here-is-what-i-learned